api/src/main/java/org/openmrs/Diagnosis.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+/**
+ * Diagnosis class defines the identification of the nature of an illness or other problem 
+ * by examination of the symptoms during an encounter (visit or interaction of a patient with a healthcare worker).
+ * 
+ * Each encounter may have 0..n diagnoses.
+ * 
+ * Every diagnosis must have a rank which must be a positive integer, and it's okay for two diagnoses 
+ * in an encounter to have the same rank.(As compared to previous EMRAPI implementation, 1=PRIMARY, 2=SECONDARY)
+ * ConditionVerificationStatus shows if the diagnosis is PROVISIONAL or CONFIRMED 
+ **/
+public class Diagnosis extends BaseChangeableOpenmrsData{
+	
+	private Integer id;
+	
+	private Encounter encounter;
+	
+	private CodedOrFreeText diagnosis;
+
+	/**
+	 * Condition 
+	 * 
+	 * May refer to a new condition to be created as a result of this diagnosis,
+	 * or an existing condition that this encounter is following up.
+	 * 
+	 * 
+	 **/
+	private Condition condition;
+
+	private ConditionVerificationStatus certainty;
+	
+	private Integer rank;
+	
+	public Diagnosis() {
+	}
+
+	/**
+	 * Diagnosis constructor with encounter, CodedOrFreeText, id, ConditionVerificationStatus, and rank
+	 *
+	 * @param encounter the encounter during which the diagnosis was made
+	 * @param diagnosis the CodedOrFreeText of the diagnosis
+	 * @param certainty the ConditionVerificationStatus of the diagnosis
+	 * @param rank the rank of the diagnosis
+	 */
+	public Diagnosis(Encounter encounter, CodedOrFreeText diagnosis, ConditionVerificationStatus certainty, Integer rank) {
+		this.encounter = encounter;
+		this.diagnosis = diagnosis;
+		this.certainty = certainty;
+		this.rank = rank;
+	}
+	
+	/**
+	 * Returns the diagnosis id.
+	 */
+	@Override
+	public Integer getId() {
+		return id;
+	}
+
+	/**
+	 * @param id
+	 * Sets diagnosis id
```

Comment:
Did you get a chance to look at how the javadoc documentation was done here? https://github.com/openmrs/openmrs-core/commit/f5132a1dda4a6a8fcbfe06a32dcbbe4b7a0a270d